File: src/main/java/me/testaccount666/serversystem/listener/executables/colorchat/ListenerColorChat.java
package me.testaccount666.serversystem.listener.executables.colorchat;

import io.papermc.paper.event.player.AsyncChatEvent;
import me.testaccount666.serversystem.ServerSystem;
import me.testaccount666.serversystem.managers.PermissionManager;
import me.testaccount666.serversystem.utils.ComponentColor;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;

public class ListenerColorChat implements Listener {
    private final boolean _enabled;

    public ListenerColorChat() {
        var config = ServerSystem.Instance.getConfigManager().getGeneralConfig();
        _enabled = config.getBoolean("Chat.ColorChat.Enabled");
    }

    @EventHandler
    public void onColorChat(AsyncChatEvent event) {
        if (!_enabled) return;
        if (!PermissionManager.hasPermission(event.getPlayer(), "Chat.ColorChat", false)) return;
        var message = ComponentColor.componentToString(event.message());

        event.message(ComponentColor.translateToComponent(message));
    }
}

File: src/main/kotlin/me/testaccount666/serversystem/listener/executables/chat/ListenerColorChat.kt
package me.testaccount666.serversystem.listener.executables.chat

import io.papermc.paper.event.player.AsyncChatEvent
import me.testaccount666.paperktx.extensions.ComponentExtensions.asComponent
import me.testaccount666.paperktx.extensions.ComponentExtensions.asString
import me.testaccount666.serversystem.extensions.getService
import me.testaccount666.serversystem.managers.PermissionManager.hasPermission
import me.testaccount666.serversystem.managers.config.ConfigurationManager
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener

class ListenerColorChat : Listener {
    private val _enabled = getService<ConfigurationManager>().generalConfig.getBoolean("Chat.ColorChat.Enabled")

    @EventHandler
    fun onColorChat(event: AsyncChatEvent) {
        if (!_enabled) return
        if (!hasPermission(event.getPlayer(), "Chat.ColorChat", false)) return
        event.message(event.message().asString().asComponent())
    }
}
